### ISSUE TITLE
Followup questions on pretraining split

### ISSUE THREAD

**Issue by mirmuss** [moment-timeseries-foundation-model/moment-research#19]
Thanks for your reply,

> I agree with you in principle. However, we define zero-shot as zero parameter updates on the testing dataset.

I find this definition a little misleading, since the general definition of zero-shot is vastly different. With this definition all the supervised models are zero shot models.

> I disagree that there is any data leakage, given the training and testing sets are disjoint.

Can you please kindly point me where the test split is being dropped in your current pipeline when args.task_name == "pre-training"?

And how is that the reopening of the issue https://github.com/moment-timeseries-foundation-model/moment-research/issues/17 is disabled? @mononitogoswami

**Comment by mononitogoswami** [comment]
> Can you please kindly point me where the test split is being dropped in your current pipeline when args.task_name == "pre-training"?

Each dataset class handles dataset splitting. There are two parameters of interest, the `data_split` parameter, which can take  3 values: "train", "val", or "test", and the `task_name` parameter, which controls to form of the dataset.

In the [base task](https://github.com/moment-timeseries-foundation-model/moment-research/blob/main/moment/tasks/base.py), the train dataloader is loaded setting `data_split` to "train" (https://github.com/moment-timeseries-foundation-model/moment-research/blob/3ab637e413f35f2c317573c0ace280d825c558de/moment/tasks/base.py#L41)

Thus, in the [dataloader file](https://github.com/moment-timeseries-foundation-model/moment-research/blob/main/moment/data/dataloader.py), the train sets of all the training datasets are concatenated. 

Does that answer your question?

> general definition of zero-shot is vastly different

Could you please provide the general definition of zero-shot that you are referring to?

**Comment by mirmuss** [comment]
> Could you please provide the general definition of zero-shot that you are referring to?

The paper you have cited in your work should be fine:

> Zero-shot TS forecasting task definition. One of the base datasets, a source dataset, is used to train a machine learning model. The trained model then forecasts a TS in a target dataset. The source and the target datasets are distinct: they do not contain TS whose values are linear transformations of each other. The forecasted TS is split into two non-overlapping pieces: the history, and the test. The history is used as model input and the test is used to compute the forecast error metric. We use the history and the test splits for the base datasets consistent with their original publication, unless explicitly stated otherwise. To produce forecasts, the model is allowed to access the TS in the target dataset on a one-at-a-time basis. This is to avoid having the model implicitly learn/adapt based on any information contained in the target dataset other than the history of the forecasted TS. If any adjustments of model parameters or hyperparameters are necessary, they are allowed exclusively using the history of the forecasted TS.

Oreshkin, B. N., Carpov, D., Chapados, N., and Bengio, Y.Meta-learning framework with applications to zero-shot time-series forecasting.Proceedings of the AAAI Conference on Artificial Intelligence, 35(10):9242–9250, May 2021.

> Thus, in the [dataloader file](https://github.com/moment-timeseries-foundation-model/moment-research/blob/main/moment/data/dataloader.py), the train sets of all the training datasets are concatenated.

Thanks a lot for this, I have missed this part
